presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrderCurveSearcher.java

```diff
@@ -129,6 +112,45 @@ public static ZOrderCurveSearcher createZOrderCurveSearcher(ZOrder zOrder, List<
         return nonNullValues;
     }
 
+    private void combineAddressRanges(List<ZAddressRange<Long>> addressRanges)
```

Comment:
An idea which might not make sense: could this be a static helper function in class `ZAddressRange`? This seems to be a generic functionality for merging ranges.

Reply:
You're right! It can definitely be static :)